### ISSUE TITLE
Give tests a finite time limit

### ISSUE THREAD

**Issue by msmith-techempower** [TechEmpower/FrameworkBenchmarks#3684]
In the latest run on Citrine, `Revenj` took 29 hours to build and would never (presumably) finish, which blocked the rest of the benchmarks.

Add in some sanity checks for build-time of tests and kill them if they take more than... what... an hour? I know that in the past `wt` used to take 50+ minutes to build. Is this still true?

**Comment by bhauer** [comment]
An hour sounds good and generous.

**Comment by nbrady-techempower** [comment]
I agree that an hour sounds good. If it goes over an hour it won't pass travis anyway, so we'll hold the PR up to make sure it can be fixed. If it goes over an hour in citrine it probably means that it hung somewhere and should be killed anyway.
